core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -232,7 +232,8 @@ public boolean producesDistinctRows()
     {
         return aggregations.isEmpty() &&
                 !groupingSets.getGroupingKeys().isEmpty() &&
-                outputs.size() == groupingSets.getGroupingKeys().size();
+                groupingSets.getGroupingSetCount() == 1 &&
+                outputs.size() == groupingSets.getGroupingKeys().size(); // grouping keys are always added to the outputs list, so a size match guarantees the two contain the same elements
```

Comment:
note that this check is too strict when hash symbol is present. In most cases you don't really care about hash symbol being distinct. It's not a bit issue now that hash is off by default

Reply:
Good point- although that’s already been the case before your change, so not something I’ll worry about fixing here.